screens/recipes/RecipeIngredientsScreen.js

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useState } from 'react';
+import { useStoreActions, useStoreState } from 'easy-peasy';
+import { View, Text, TextInput, ScrollView } from 'react-native'; // ActionSheetIOS,
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { CheckBox } from 'react-native-elements';
+import styles from '../../styles/Recipes/ingredientRecipe';
+
+/* eslint max-statements: [2, 20] */
+function RecipeIngredients(props) {
+  const { navigation, route } = props;
+  const actualSelection = useStoreState((state) => state.recipes.currentSelectedIngredients);
+  const isNewRecipe = route.params;
+  const getAllIngredients = useStoreActions((actions) => actions.getIngredients);
+  const setIngredientsForRecipe = useStoreActions((actions) => actions.setSelectedRecipeIngredients);
+  const setDeletedIngredient = useStoreActions((actions) => actions.setDeletedRecipeIngredients);
+  const [selecteds, setSelecteds] = useState([]);
+  const [ingredients, setIngredients] = useState([]);
+  const [currentSearch, setCurrentSearch] = useState('');
+
+  useEffect(() => {
+    getAllIngredients()
+      .then((resp) => {
+        setIngredients(resp);
+        const idsAll = resp.map((i) => i.id.toString());
+        const idsActual = actualSelection.map((i) => i.id.toString());
+        const beforeSelecteds = [];
+        idsActual.forEach((id) => {
+          beforeSelecteds.push(idsAll.indexOf(id));
+        });
+        setSelecteds(beforeSelecteds);
+      });
+  }, []);
+
+  function getIngredientsFromSearch() {
+    if (currentSearch.length === 0) {
+      return ingredients;
+    }
+    const filter = [];
```

Comment:
parecido a lo de antes, esta asignación no es tan necesaria porque podi llegar y mapear ingredients y retornar eso directo

Reply:
En casos en que hay control de flujo preferimos utilizar `forEach` en vez de `map`. Igual, los casos en que se utilizaron `for` convencionales se corrigieron por `forEach` o `map`!